core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2474,25 +2478,42 @@ private[spark] class CallerContext(
    val context = "SPARK_" + from + appIdStr + appAttemptIdStr +
      jobIdStr + stageIdStr + stageAttemptIdStr + taskIdStr + taskAttemptNumberStr
 
+  lazy val conf = new Configuration
+
   /**
    * Set up the caller context [[context]] by invoking Hadoop CallerContext API of
    * [[org.apache.hadoop.ipc.CallerContext]], which was added in hadoop 2.8.
    */
   def setCurrentContext(): Boolean = {
-    var succeed = false
-    try {
-      // scalastyle:off classforname
-      val callerContext = Class.forName("org.apache.hadoop.ipc.CallerContext")
-      val Builder = Class.forName("org.apache.hadoop.ipc.CallerContext$Builder")
-      // scalastyle:on classforname
-      val builderInst = Builder.getConstructor(classOf[String]).newInstance(context)
-      val hdfsContext = Builder.getMethod("build").invoke(builderInst)
-      callerContext.getMethod("setCurrent", callerContext).invoke(null, hdfsContext)
-      succeed = true
-    } catch {
-      case NonFatal(e) => logInfo("Fail to set Spark caller context", e)
+    if (!CallerContext.callerContextSupported) {
```

Comment:
The point is to avoid a bunch of exception stack traces in the log over and over. I don't think thread-safety is an issue here. If it prints twice instead of once, no big deal. Otherwise this error/warning prints on every task execution. That's not great.